lib/internal/modules/cjs/loader.js

```diff
@@ -816,11 +816,11 @@ Module._resolveLookupPaths = function(request, parent) {
     return paths.length > 0 ? paths : null;
   }
 
-  // With --eval, parent.id is not set and parent.filename is null.
+  // In REPL, parent.filename is null.
   if (!parent || !parent.id || !parent.filename) {
     // Make require('./path/to/foo') work - normally the path is taken
-    // from realpath(__filename) but with eval there is no filename
-    const mainPaths = ['.'].concat(Module._nodeModulePaths('.'), modulePaths);
+    // from realpath(__filename) but in REPL there is no filename
+    const mainPaths = ['.'];
```

Comment:
With `--eval`, `parent` has both `id` and `filename` set now, so I think this edge case wasn't for `--eval` anymore but for `REPL` mode.
```sh
> node --eval 'console.log(module)'
Module {
  id: '[eval]',
  path: '.',
  exports: {},
  parent: undefined,
  filename: '/Users/zyszys/Projects/nodejs/node/[eval]',
  loaded: false,
  children: [],
  paths: [
    '/Users/zyszys/Projects/nodejs/node/node_modules',
    '/Users/zyszys/Projects/nodejs/node_modules',
    '/Users/zyszys/Projects/node_modules',
    '/Users/zyszys/node_modules',
    '/Users/node_modules',
    '/node_modules'
  ]
}
```